blocks/large-manual-promo-block/features/large-manual-promo/default.jsx

```diff
@@ -1,19 +1,112 @@
 import React from "react";
 import PropTypes from "@arc-fusion/prop-types";
-import { useFusionContext } from "fusion:context";
-import { LazyLoad, isServerSide } from "@wpmedia/engine-theme-sdk";
+import { useComponentContext, useFusionContext } from "fusion:context";
+import { useEditableContent } from "fusion:content";
+import { LazyLoad } from "@wpmedia/engine-theme-sdk";
+import {
+	formatURL,
+	isServerSide,
+	Overline,
+	MediaItem,
+	Grid,
+	Link,
+	Heading,
+	HeadingSection,
+	Image,
+	Paragraph,
+	Stack,
+} from "@wpmedia/arc-themes-components";
 import { imageRatioCustomField } from "@wpmedia/resizer-image-block";
-import { LargePromoPresentation } from "@wpmedia/shared-styles";
+
+const BLOCK_CLASS_NAME = "b-large-manual-promo";
 
 const LargeManualPromo = ({ customFields }) => {
 	const { isAdmin } = useFusionContext();
-	if (customFields?.lazyLoad && isServerSide() && !isAdmin) {
+	const {
+		headline,
+		imagePosition,
+		imageURL,
+		lazyLoad,
+		caption,
+		credits,
+		subtitle,
+		linkURL,
+		newTab,
+		showHeadline,
+		showImage,
+		description,
+		overline,
+		showOverline,
+		showDescription,
+		overlineURL,
+		hideImageCaption,
+		hideImageCredits,
+		hideImageTitle,
+	} = customFields;
+	const shouldLazyLoad = lazyLoad && !isAdmin;
+	const { registerSuccessEvent } = useComponentContext();
+
+	if (shouldLazyLoad && isServerSide()) {
 		// On Server
 		return null;
 	}
+
+	const PromoImage = () => {
+		const { searchableField } = useEditableContent();
+
+		return showImage ? (
+			<MediaItem
+				caption={!hideImageCaption ? caption : null}
+				credit={!hideImageCredits ? credits : null}
+				title={!hideImageTitle ? subtitle : null}
+				{...searchableField("imageURL")}
+				suppressContentEditableWarning
+			>
+				<Image alt={headline} src={imageURL} searchableField />
+			</MediaItem>
+		) : null;
+	};
+
+	const PromoHeading = () =>
+		showHeadline ? (
+			<Heading>
+				{linkURL ? (
+					<Link href={formatURL(linkURL)} openInNewTab={newTab} onClick={registerSuccessEvent}>
+						{headline}
+					</Link>
+				) : (
+					headline
+				)}
+			</Heading>
+		) : null;
+
+	const PromoOverline = () => {
+		if (showOverline) {
+			if (overlineURL) {
+				return <Overline href={overlineURL}>{overline}</Overline>;
+			}
+			return <Overline>{overline}</Overline>;
+		}
+		return null;
+	};
+
+	const PromoDescription = () => (showDescription ? <Paragraph>{description}</Paragraph> : null);
+
 	return (
-		<LazyLoad enabled={customFields?.lazyLoad && !isAdmin}>
-			<LargePromoPresentation {...customFields} />
+		<LazyLoad enabled={shouldLazyLoad}>
+			<HeadingSection>
+				<Grid
+					role="article"
+					className={`${BLOCK_CLASS_NAME} ${BLOCK_CLASS_NAME}__${imagePosition}`}
```

Comment:
The image position class is a modifier we should use the BEM syntax hyphens `--` 
```suggestion
					className={`${BLOCK_CLASS_NAME} ${BLOCK_CLASS_NAME}--${imagePosition}`}
```

Reply:
I don't see a CSS class in the Sass file for this. Is this needed?

Reply:
Removed it.  Had to keep `BLOCK_CLASS_NAME` for gap. Just removed `BLOCK_CLASS_NAME}__${imagePosition}`